File: chrome-extension/icons/generate-icons.py
#!/usr/bin/env python3
"""
生成 Chrome 擴充功能所需的圖標
需要安裝 Pillow: pip install Pillow
"""

try:
    from PIL import Image, ImageDraw
    import os

    def generate_icon(size):
        """生成指定尺寸的圖標"""
        # 創建新圖片，RGBA 模式支援透明
        img = Image.new('RGB', (size, size), color='#667eea')
        draw = ImageDraw.Draw(img)
        
        # 繪製漸層背景（簡單版本：使用兩個顏色）
        # 從左上到右下的漸層效果
        for i in range(size):
            # 計算漸層顏色
            ratio = i / size
            r1, g1, b1 = 0x66, 0x7e, 0xea  # #667eea
            r2, g2, b2 = 0x76, 0x4b, 0xa2  # #764ba2
            r = int(r1 + (r2 - r1) * ratio)
            g = int(g1 + (g2 - g1) * ratio)
            b = int(b1 + (b2 - b1) * ratio)
            color = (r, g, b)
            draw.line([(i, 0), (i, size)], fill=color)
        
        # 繪製書本圖標（兩個重疊的矩形）
        book_width = int(size * 0.6)
        book_height = int(size * 0.4)
        x = (size - book_width) // 2
        y = (size - book_height) // 2
        
        # 左頁
        draw.rectangle([x, y, x + book_width // 2, y + book_height], fill='white')
        # 右頁
        draw.rectangle([x + book_width // 2, y, x + book_width, y + book_height], fill='white')
        # 書脊線
        draw.line([(x + book_width // 2, y), (x + book_width // 2, y + book_height)], 
                  fill='#667eea', width=max(1, size // 32))
        
        return img

    # 生成三個尺寸的圖標
    sizes = [16, 48, 128]
    
    for size in sizes:
        icon = generate_icon(size)
        filename = f'icon{size}.png'
        icon.save(filename, 'PNG')
        print(f'✅ 已生成 {filename} ({size}x{size})')
    
    print('\n🎉 所有圖標已生成完成！')

except ImportError:
    print('❌ 需要安裝 Pillow 套件')
    print('💡 請執行: pip install Pillow')
    print('\n或者使用瀏覽器打開 generate-icons.html 來生成圖標')

File: chrome-extension/icons/generate-icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

// 如果沒有 canvas，使用簡單的 SVG 轉換
function generateIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // 背景漸層
  const gradient = ctx.createLinearGradient(0, 0, size, size);
  gradient.addColorStop(0, '#667eea');
  gradient.addColorStop(1, '#764ba2');
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, size, size);

  // 繪製書本圖標（使用簡單的幾何圖形）
  ctx.fillStyle = 'white';
  ctx.strokeStyle = 'white';
  ctx.lineWidth = size * 0.05;

  // 繪製書本（兩個重疊的矩形）
  const bookWidth = size * 0.6;
  const bookHeight = size * 0.4;
  const x = (size - bookWidth) / 2;
  const y = (size - bookHeight) / 2;

  // 左頁
  ctx.fillRect(x, y, bookWidth * 0.5, bookHeight);
  // 右頁
  ctx.fillRect(x + bookWidth * 0.5, y, bookWidth * 0.5, bookHeight);
  // 書脊
  ctx.beginPath();
  ctx.moveTo(x + bookWidth * 0.5, y);
  ctx.lineTo(x + bookWidth * 0.5, y + bookHeight);
  ctx.stroke();

  return canvas.toBuffer('image/png');
}

// 生成三個尺寸的圖標
const sizes = [16, 48, 128];

sizes.forEach(size => {
  try {
    const buffer = generateIcon(size);
    fs.writeFileSync(`icon${size}.png`, buffer);
    console.log(`✅ 已生成 icon${size}.png`);
  } catch (error) {
    console.error(`❌ 生成 icon${size}.png 失敗:`, error.message);
    console.log('💡 提示: 請先安裝 canvas: npm install canvas');
  }
});
